docs/tutorial/workers.md

```diff
@@ -126,95 +123,132 @@ From our previous steps, we now have:
 
-In your terminal (not the terminal in which the worker is running), navigate to your `repo_info.py` file that we created in the first section.
-For best results, this file should be in its own otherwise _empty directory_. 
-Now run the following command ***from the root of this directory*** to begin deploying your flow:
+The updates that you need to make to `repo_info.py` are:
+
+1. Changing `flow.serve` to `flow.deploy`
+2. Tell `flow.deploy` which work pool to deploy to
+3. Tell `flow.deploy` the name to use for the Docker image it builds.
+
+Here's what the updated `repo_info.py` looks like:
+
+```python hl_lines="17-22" title="repo_info.py"
+import httpx
+from prefect import flow
+
+
+@flow(log_prints=True)
+def get_repo_info(repo_name: str = "PrefectHQ/prefect"):
+    url = f"https://api.github.com/repos/{repo_name}"
+    response = httpx.get(url)
+    response.raise_for_status()
+    repo = response.json()
+    print(f"{repo_name} repository statistics 🤓:")
+    print(f"Stars 🌠 : {repo['stargazers_count']}")
+    print(f"Forks 🍴 : {repo['forks_count']}")
+
+
+if __name__ == "__main__":
+    get_repo_info.deploy(
+        name="my-first-deployment", 
+        work_pool_name="my-docker-pool", 
+        image="my-first-deployment-image:tutorial"
+        skip_push=True
+    )
+```
+
+!!! note "Why the `skip_push=True`?"
+    For this tutorial, your Docker worker is running on your machine, so we don't need to push the image built by `flow.deploy` to a registry. When your worker is running on a remote machine, you will need to push the image to a registry that the worker can access.
+
+    You remove the `skip_push=True` argument and include your registry name to push the image to a registry.
+
+Now that you've updated your script, you can run it to deploy your flow to the work pool:
 
 <div class="terminal">
 ```bash
-prefect deploy
+python repo_info.py
 ```
 </div>
 
-!!! tip "Specifying an entrypoint"
-    In non-interactive settings (like CI/CD), you can specify the entrypoint of your flow directly in the CLI. 
-    
-    For example, if `get_repo_info` is defined in `repo_info.py`, provide deployment details with flags `prefect deploy repo_info.py:get_repo_info -n my-deployment -p my-docker-pool`.
+Prefect will build a custom Docker image containing your workflow code that the worker can use to dynamically spawn Docker containers whenever this workflow needs to run. 
 
-When running `prefect deploy` interactively, the CLI will discover all flows in your working directory. 
-Select the flow you want to deploy, and the deployment wizard will walk you through the rest of the deployment creation process:
+!!! note "What Dockerfile?"
+    In this example, Prefect generates a Dockerfile for you that will build an image based off of one of Prefect's published images. The generated Dockerfile will copy the current directory into the Docker images and install any dependencies listed in a `requirements.txt` file.
```

Comment:
Might be good to link the published Prefect images section of the docs, but I've got an open PR to move that to the guides ATM.

Reply:
Cool, I can add that link once https://github.com/PrefectHQ/prefect/pull/10910 is merged.